aten/src/ATen/native/cuda/Loops.cuh

```diff
@@ -120,6 +133,8 @@ at::opmath_type<f_inputs_type> scalar_val=0) {
   static_assert((!std::is_same<f_inputs_type, c10::complex<double>>::value &&
   !std::is_same<return_type, c10::complex<float>>::value), "complex types are not supported \
   in jiterator functors");
+  static_assert(
+      sizeof...(extra_args) < 7, "more than 7 captures is not allowed!");
```

Comment:
Arbitrary

Reply:
How much of an impact does this number have?

Because of the way we specialize these templates per function (because the string name of the function is a template arg) can this pass the # extra args as a template param, too, so a maximum size for such args doesn't need to be set?